jupyter_book/pdf.py

```diff
@@ -44,18 +68,125 @@ async def _html_to_pdf(html_file, pdf_file):
+#         "title",
+#         "author",
+#         "theme",
+#         "toctree_only",
+#     )
+#     updated = list(copy(latex_document))
+#     for i, (_, name) in enumerate(zip(latex_document, names)):
+#         if name in updates:
+#             updated[i] = updates[name]
+#     return tuple(updated)
+
+
+def update_latex_documents(latex_documents, latexoverrides):
+    """
+    Apply latexoverrides from _config.yml to latex_documents tuple
+    """
+
+    if len(latex_documents) > 1:
+        _message_box(
+            "Latex documents specified as a multi element list in the _config",
+            "This suggests the user has made custom settings to their build",
+            "[Skipping] processing of automatic latex overrides",
+        )
+        return latex_documents
+
+    # Extract latex document tuple
+    latex_document = latex_documents[0]
+    # Apply single overrides from _config.yml
+    updated_latexdocs = []
+    for loc, item in enumerate(LATEX_DOCUMENTS):
+        # the last element toctree_only seems optionally included
+        if loc >= len(latex_document):
+            break
+        if item in latexoverrides.keys():
+            updated_latexdocs.append(latexoverrides[item])
+        else:
+            updated_latexdocs.append(latex_document[loc])
+    return [tuple(updated_latexdocs)]
+
+
+def latex_document_components(latex_documents):
+    """ Return a dictionary of latex_document components by name """
+    latex_tuple_components = {}
+    for idx, item in enumerate(LATEX_DOCUMENTS):
+        # skip if latex_documents doesn't doesn't contain all elements
+        # of the LATEX_DOCUMENT specification tuple
+        if idx >= len(latex_documents):
+            continue
+        latex_tuple_components[item] = latex_documents[idx]
+    return latex_tuple_components
+
+
+def latex_document_tuple(components):
+    """ Return a tuple for latex_documents from named components dictionary """
+    latex_doc = []
+    for item in LATEX_DOCUMENTS:
+        if item not in components.keys():
+            continue
+        else:
+            latex_doc.append(components[item])
+    return tuple(latex_doc)
+
+
+def autobuild_singlepage_latexdocs(app):
+    """
+    Build list of tuples for each document in the Project
+
+    [((startdocname, targetname, title, author, theme, toctree_only))]
+
+    https://www.sphinx-doc.org/en/3.x/usage/configuration.html#confval-latex_documents
+    """
+    latex_documents = app.config.latex_documents
+    if len(latex_documents) > 1:
+        _message_box(
+            "Latex documents specified as a multi element list in the _config",
+            "This suggests the user has made custom settings to their build",
+            "[Skipping] --individualpages option",
+        )
+        return latex_documents
+
+    # Extract latex_documents updated tuple
+    latex_documents = latex_documents[0]
+
+    titles = app.env.titles
+    master_doc = app.config.master_doc
+    if "/" in master_doc:
```

Comment:
this seems like it would fail on Windows?

Reply:
@chrisjsewell good pickup -- do we have any CI for windows?

Reply:
We run the whole test suite against Windows, but I imagine scenario is not yet covered (it probably should be)

Reply:
thanks @chrisjsewell that's what I though --I will see if we can add a test.

Reply:
thanks @chrisjsewell for this pickup. I have simplified the code as `os.path.dirname` returns sensible values for all path cases in 
9a0d16f